File: pocketr-java.api/src/main/java/com/decrux/pocketr/api/services/OwnershipGuard.java
package com.decrux.pocketr.api.services;

import com.decrux.pocketr.api.exceptions.ForbiddenException;
import java.util.Objects;
import org.springframework.stereotype.Component;

@Component
public class OwnershipGuard {

    public void requireOwner(Long resourceOwnerId, long actorId) {
        requireOwner(resourceOwnerId, actorId, "Access denied");
    }

    public void requireOwner(Long resourceOwnerId, long actorId, String message) {
        if (!Objects.equals(resourceOwnerId, actorId)) {
            throw new ForbiddenException(message);
        }
    }
}

File: pocketr-api/src/main/kotlin/com/decrux/pocketr/api/services/OwnershipGuard.kt
package com.decrux.pocketr.api.services

import com.decrux.pocketr.api.exceptions.ForbiddenException
import org.springframework.stereotype.Component

@Component
class OwnershipGuard {
    fun requireOwner(
        ownerUserId: Long?,
        requestingUserId: Long,
        message: String = "Access denied",
    ) {
        if (ownerUserId != requestingUserId) throw ForbiddenException(message)
    }
}
